src/Symfony/Bridge/Doctrine/Tests/PropertyInfo/DoctrineExtractorTest.php

```diff
@@ -165,7 +165,7 @@ public static function typesProvider(): array
         return [
             ['id', [new Type(Type::BUILTIN_TYPE_INT)]],
             ['guid', [new Type(Type::BUILTIN_TYPE_STRING)]],
-            ['bigint', [new Type(Type::BUILTIN_TYPE_STRING)]],
+            ['bigint', [new Type(Type::BUILTIN_TYPE_STRING)]], // change? separate for DBAL 3 vs 4?
```

Comment:
I suppose yes

Reply:
either separate tests, or just an expectation that changes depending on which version is installed - that'd be fine also

Reply:
Hey Nicolas, 

as Alexander hinted, you can use composer to [get installed versions](https://getcomposer.org/doc/07-runtime.md#knowing-whether-package-x-is-installed-in-version-y). Do you think this is worth it? Or should I just update the test with a new return expected value?

Reply:
Let's rather go with a feature test. There must be some class that's unique to one or the other version.

Reply:
For instance, check if the `\Doctrine\DBAL\Tools\Console\ConsoleRunner` class exists. If it is, we're on DBAL <4.

Reply:
You can check if the `BigIntType` has a `getName()` method. That should not be the case on DBAL 4.

Reply:
~Ok, I actually even thought about having 2 extractors, one per DBAL major, and then in the compiler pass to set the respective class with the class exists check~. 

I will do class exists as it is simple and straightforward, composer checks are fine, but, it implies composer is there to begin with 🤷

Reply:
> it implies composer is there to begin with 🤷

That does not sound like a strong requirement. 😄 

That being said, on this repository, we rather do feature detection than relying on Composer's runtime API, so `method_exists()` is the way to go here.

Reply:
@nicolas-grekas I updated the PR with Alexander's hint/tip/?order. If there are still changes I'd need to apply let me know :+1: